pkg/operator/controller/dns/controller.go

```diff
@@ -449,11 +449,14 @@ func mergeConditions(conditions, updates []iov1.DNSZoneCondition) []iov1.DNSZone
 		for j, cond := range conditions {
 			if cond.Type == update.Type {
 				add = false
-				if conditionChanged(cond, update) {
+				statusChanged, changes := controller.ConditionChanged(conditions[j].Message, update.Message, string(conditions[j].Status), string(update.Status), conditions[j].Reason, update.Reason)
```

Comment:
The variable is named `statusChanged`, but its value depends on whether status _or_ message or reason changed.  

To fix OCPBUGS-37491, it should be _sufficient_ to check status alone, and besides that, it would be more _correct_ to check status alone.  The discussion at https://github.com/openshift/cluster-ingress-operator/pull/465#discussion_r494715168 addresses correctness.  As for sufficiency, here is my reasoning:

1. The canary controller observes that it cannot connect to the canary route, and so the controller sets `CanaryChecksSucceeding=False`.
2. `computeIngressDegradedCondition` observes that the `CanaryChecksSucceeding` status condition has been updated within the past 60 seconds, based on its `lastTransitionTime`, so the function ignores the condition.
3. The canary controller checks again and observes that it cannot connect to the canary route, and it updates the _message_ for the `CanaryChecksSucceeding` condition.  Because of the incorrect logic in `mergeConditions`, this also updates `lastTransitionTime`.  
4. `computeIngressDegradedCondition` again observes that the status condition's `lastTransitionTime` is within the last 60 seconds, so it again ignores the status condition.

And repeat _ad infinitum_.

So, if I understand correctly, if we simply set `lastTransitionTime` correctly to reflect the time when `CanaryChecksSucceeding`  _transitioned state_ (in this case, from `True` to `False`), then `computeIngressDegradedCondition` would correctly set `Degraded=True` the second time it observed the `CanaryChecksSucceeding=False` condition, 60 seconds after that transition.

Reply:
There are two variables: `statusChanged`, and `changes`.  The former records if the status changed and he latter records if anything changed.  So this does work.  I will make the changes to tighten this up.

Reply:
@Miciah It works without all the extra baggage, and I uploaded the change.  PTAL when you get a chance.